src/graphchain/core.py

```diff
@@ -349,32 +359,32 @@ def optimize(
     keys: Optional[Union[Hashable, Iterable[Hashable]]] = None,
     skip_keys: Optional[Container[Hashable]] = None,
     location: Union[str, fs.base.FS, CacheFS] = "./__graphchain_cache__",
+    serialize: Callable[[Any, io.BytesIO], None] = joblib_dump_lz4,
```

Comment:
There may be issues with assuming any serializer can work with a file descriptor. For instance, dask's own `DataFrame.to_parquet` takes a **directory** and not a file descriptor, because it needs to be able to write out each partition to a separate file. Would it be okay to just pass the PyFilesystem path and the filesystem `cache_fs`, so that users can do what they want, with a docstring explaining that `self.cache_fs.open(fn, "rb")` is often what they will want to do?

Reply:
Done. The serialize/deserialize API now gives you control over which files you write to the filesystem (if any). This is the default implementation that would come with graphchain:

```python
def joblib_dump(obj: Any, fs: fs.base.FS, key: str, **kwargs: Dict[str, Any]) -> None:
    """Store an object on a filesystem."""
    filename = f"{key}.joblib"
    if isinstance(kwargs.get("compress"), tuple) and len(kwargs["compress"]) == 2:
        filename = f"{filename}.{kwargs['compress'][0]}"  # type: ignore[index]
    elif isinstance(kwargs.get("compress"), str):
        filename = f"{filename}.{kwargs['compress']}"
    with fs.open(filename, "wb") as fid:
        joblib.dump(obj, fid, **kwargs)


def joblib_load(
    fs: fs.base.FS, key: str, filename_extension: str = "joblib", **kwargs: Dict[str, Any]
) -> Any:
    """Load an object from a filesystem."""
    filename = f"{key}.{filename_extension}"
    with fs.open(filename, "rb") as fid:
        return joblib.load(fid, **kwargs)

joblib_dump_lz4 = partial(joblib_dump, compress="lz4", protocol=HIGHEST_PROTOCOL)
joblib_load_lz4 = partial(joblib_load, filename_extension="joblib.lz4")
```